tests/test_twisted.py

```diff
@@ -40,6 +42,7 @@ def test_reports_metrics(self):
         url = f"http://localhost:{port}/metrics"
         d = agent.request(b"GET", url.encode("ascii"))
 
+        filterwarnings('ignore', category=DeprecationWarning)
```

Comment:
I believe this needs to be inside a `with warnings.catch_warnings():` block so that we don't filter deprecation warnings coming from other parts of the code?

Reply:
I tried catching warnings using `catch_warnings` but without success.

```python
with warnings.catch_warnings():
    warnings.filterwarnings("ignore", category=DeprecationWarning)
    d.addCallback(readBody)
```

However, I think the filtered warning can be limited by adding the `message` parameter.

Do you accept this modification?

```python
filterwarnings("ignore", category=DeprecationWarning, message="Using readBody "
               "with a transport that does not have an abortConnection method")
```

Reply:
Yep, that sounds good to me. Thanks!